pire/scanners/slow.h

```diff
@@ -248,9 +249,12 @@ class SlowScanner {
 		}
 	}
 
-	explicit SlowScanner(Fsm& fsm, bool needActions = false, bool removeEpsilons = true)
+	explicit SlowScanner(Fsm& fsm, size_t distance = 0, bool needActions = false, bool removeEpsilons = true)
```

Comment:
I would suggest to put `distance` to the last place, to ensure backward compatibility.